### PR TITLE
Remember main window position for reappearing window

### LINKED ISSUE #31: Window position not remembered?
I like to move the screenshot window into a corner, so that it doesn't cover up the subject that I'd like to take a screenshot of. (I know the window is hidden automatically so this doesn't really matter). But when the screenshot is taken and saved the screenshot window is opened again in the middle of the screen. The previous window location isn't remembered / restored.

### PR DESCRIPTION
**Pull request by purejava**
This is intended to resolve #31.

### CODE REVIEW

File: src/ScreenshotWindow.vala [hunk 1]
```diff
@@ -560,5 +569,11 @@ namespace Screenshot {
         private void cancel_clicked () {
             destroy ();
         }
+
+        // Save main window position so that this position can be used
+        // when the window reappears again
+        private void remember_window_position () {
```
Comment (donadigo) [inline src/ScreenshotWindow.vala]: Please check if the `get_window ()` doesn't return `null` so you don't fail calling `get_root_origin`.

File: src/ScreenshotWindow.vala [hunk 2]
```diff
@@ -560,5 +569,14 @@ namespace Screenshot {
...
+	    Gdk.Window window = this.get_toplevel ().get_window ();
```
Comment (donadigo) [inline src/ScreenshotWindow.vala]: Maybe use `var` here? If you want to use the type explicitly this should be `Gdk.Window?` not `Gdk.Window` with the `?` to make it nullable.

**Comment by purejava** [comment]
You are right of course.
I'll go with `var` and change this later.

**Comment by purejava** [comment]
Thanks for your advices.

**Comment by peteruithoven** [comment]
@purejava thanks!

**Pull request merged**